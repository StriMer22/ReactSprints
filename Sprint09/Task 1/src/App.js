import React from 'react';
import './App.css';

export default function App() {
    let key = 1
    const list = `Animals
        Anime
        Anti-Malware
        Art Design
        Books
        Business
        Calendar
        Cloud Storage
        File Sharing
        Animals
        Continuous Integration
        Cryptocurrency`
    const arr = list.split('\n        ').map(el => {
        key++
        return <li key={key}>{el}</li>
    })
    return (
        <div>
            Some List:
            <ul>
                {arr}
            </ul>
        </div>
    );
}
