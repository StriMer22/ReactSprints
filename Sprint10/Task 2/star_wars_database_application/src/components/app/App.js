import './App.module.css';
import {get, getImg} from "../../services/sw-service";
import React, {useEffect, useState} from "react";
import {Navbar} from "../Navbar/Navbar";
import {Main} from "../Main/Main";

let count = 1
let query = 'people'

function App() {

    const [useData, setData] = useState('')
    const [useImg, setImg] = useState('')

    useEffect(() => {
        get(query, count).then(res => {
            setData(res)
        })
        getImg(query, count).then(res => setImg(res))
    }, [setData])

    function changeQuery(event) {
        query = event.target.innerHTML.toLowerCase()
        count = 1
        get(query, count).then(res => {
            setData(res)
        })
        getImg(query, count).then(res => setImg(res))
    }

    function add() {
        count++
        get(query, count).then(res => {
            setData(res)
        })
        getImg(query, count).then(res => setImg(res))
    }

    return (
        <>
           <Navbar changeQuery={changeQuery} />

            <button onClick={add}>NEXT</button>

            <Main useData={useData} useImg={useImg} query={query}/>
        </>)
}
export default App;