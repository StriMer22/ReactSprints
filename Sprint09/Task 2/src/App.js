import React from 'react';
import './App.css';
import First from "./First";

export default function App() {

    const list = `Animals
        Anime
        Anti-Malware
        Art Design
        Books
        Business
        Calendar
        Cloud Storage
        File Sharing
        Animals
        Continuous Integration
        Cryptocurrency`
    const arr = list.split('\n        ').map(el=>el.toLowerCase())

    return (
        <div>
            Some List:
            <First content={arr} />
        </div>
    );
}