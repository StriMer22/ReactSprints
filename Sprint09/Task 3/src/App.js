import React from 'react';
import './App.css';

export default class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {appData : `React Marathon`}
    }


    changeSize = () => {
        this.setState((state)=>{
            return {appData: state.appData.toLowerCase()}
        })
    }

    render() {
        return <div onClick={this.changeSize}> {this.state.appData} </div>
    }
}
