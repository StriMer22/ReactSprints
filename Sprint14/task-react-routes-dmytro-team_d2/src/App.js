import React, { useState } from 'react';

import './App.css';

import { SubTask1 } from './SubTask1';
import { SubTask2 } from './SubTask2';
import { SubTask3 } from './SubTask3';
import { SubTask4 } from './SubTask4';

import { BrowserRouter as Router, Switch, Route, Link, Redirect } from 'react-router-dom';

export default function App() {
  const [even, setEven] = useState(false);
  const [check, setCheck] = useState(false);
  return (
    <Router>
      <div className='App'>
        <h1>React Marathon</h1>
        <h2>The topic 'Routes'</h2>
        <Link to="/subtask3">Show protected information if</Link>
        <span>&nbsp;</span>
        <input onChange={(e) => setEven(e.target.value %2 !== 0)} size="5"></input> is odd
        <div className='mainClass'>
          Go to the component programmatically, by checking the box:{' '}
          <input onChange={(event) => setCheck(event.target.checked)} type='checkbox'></input>
        </div>
        <Switch>
          <Route path='/subtask1/:id' component={SubTask1}></Route>
          <Route path="/subtask2" component={SubTask2}/>
          <Route path="/subtask3"><SubTask3 even={even}/></Route>
          <Route exact path='/' render={()=>check ? (<Redirect to='/subtask4'/>) : (<Redirect to='/'/>)}/>
          <Route path="/subtask4"><SubTask4 check={check}/></Route>
        </Switch>
      </div>
    </Router>
  );
}