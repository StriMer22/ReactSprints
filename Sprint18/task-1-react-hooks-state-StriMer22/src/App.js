import './App.css';
import {useState} from "react";

export default function App() {

  const [data,setData] = useState('React Marathon')

  return <div onClick={()=>{
  return setData(data.toLowerCase())
  }}> {data} </div>;
}