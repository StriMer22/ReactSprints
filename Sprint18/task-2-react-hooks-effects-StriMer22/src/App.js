import './App.css';
import {useEffect, useState} from "react";

export default function App() {

    const[value,setValue] = useState()

    useEffect(()=>{
        setValue(localStorage.getItem('appData'))
    },[])

  return (
    <div>
      React Marathon, appData: <input size='5' value={value}/>
    </div>
  );
}