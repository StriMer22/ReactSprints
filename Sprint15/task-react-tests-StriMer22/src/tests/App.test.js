import DataService from '../dataService';
import App from '../App';
import reactDom from '../index';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import React from 'react';
jest.mock('react-dom', () => ({ render: jest.fn() }));

describe('Sprint 14 task', () => {
  test('Test dataService', () => {
    new DataService().getGitHubUser().then(data => {
      return expect({}).toEqual(data);
    });
  });
  test('Test App.js', () => {
    expect(renderer.create(<App />)).toMatchSnapshot();
  });
});
