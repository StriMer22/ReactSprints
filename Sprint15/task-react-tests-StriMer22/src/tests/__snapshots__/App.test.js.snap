// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sprint 14 task Test App.js 1`] = `
<div>
  <h3>
    GitHub User Info
  </h3>
  <ul />
</div>
`;
